### ISSUE TITLE
weird behaviour (yield related)

### ISSUE THREAD

**Issue by alinalexa** [jshint/jshint#3037]
![Image](https://ibin.co/w800/2wcvWbKdlS3S.png)
vs 
![Image](https://ibin.co/2wcwGUt4C6Yg.png)
vs
![Image](https://ibin.co/2wcxavGvzHv5.png)

`function a() {}` can be anything

`.jshintrc`
```
{
    "maxerr": 10,
    "curly": false,
    "eqeqeq": true,
    "latedef": true,
    "undef": true,
    "unused": "last",
    "maxlen": false,
    "varstmt": true,
    "asi": true,
    "esversion": 6,
    "lastsemic": true,
    "noyield": true,
    "node": true
}
```

**Comment by jugglinmike** [comment]
Confirmed. Here's a furher reduced test case:

```js
// jshint esversion: 6, asi: true
function* g() {
  void 0
  yield;
}
```

Thanks for the report

**Comment by jugglinmike** [comment]
This bug extends beyond the `yield` operator--it can also be reproduced using the increment/decrement operators, as in:

```js
// jshint asi: true
x++
void 0
```

**Comment by jugglinmike** [comment]
I've submitted a fix here: https://github.com/jshint/jshint/pull/3045. This was much trickier than it looked!

**Comment by jugglinmike** [comment]
Resolved via gh-3045. You can expect this fix to be available in the next release of JSHint (post 2.9.3).
